client/app/host/page.tsx

```diff
@@ -1,3 +1,476 @@
+        setTeamList((prev) =>
+            prev.map((team) => (team.name === teamName ? { ...team, status: false } : team))
+        );
+    };
+
+    const handleRemoveTeam = (teamName: string) => {
+        setTeamList((prev) => prev.filter((team) => team.name !== teamName));
+    };
+
+    const handleRemoveQuestion = (q: string) => {
+        setQuestions((prev) => prev.filter((question) => question.question !== q));
+    };
+
+    const handleToggleServer = () => {
+        setIsServerOn((prev) => (prev === null ? true : !prev));
+        disconnectAllTeams();
+    };
+
+    return (
+        <ResizablePanelGroup direction="horizontal" className="flex flex-grow">
+            <ResizablePanel
+                className="flex flex-col p-6"
+                defaultSize={30}
+                minSize={25}
+                maxSize={40}
+            >
+                <span className="flex h-fit w-full items-center justify-evenly">
+                    <Dialog>
+                        <DialogTrigger>
+                            <Plus />
+                        </DialogTrigger>
+                        <DialogContent>
+                            <DialogHeader>
+                                <DialogTitle>New Team</DialogTitle>
+                                <DialogDescription>
+                                    Enter a name and password for the new team. Remember these as
+                                    the team will need them to connect!
+                                </DialogDescription>
+                            </DialogHeader>
+                            <div>
+                                <Label htmlFor="name">Name</Label>
+                                <Input
+                                    id="name"
+                                    placeholder="Team Name"
+                                    value={newTeamName}
+                                    onChange={(e) => setNewTeamName(e.target.value)}
+                                />
+                                <Label htmlFor="password">Password</Label>
+                                <Input
+                                    id="password"
+                                    placeholder="Team Password"
+                                    value={newTeamPassword}
+                                    onChange={(e) => setNewTeamPassword(e.target.value)}
+                                />
+                            </div>
+                            {errorMessage && (
+                                <div className="mt-2 text-red-500">{errorMessage}</div>
+                            )}
+                            <DialogFooter>
+                                <Button type="submit" onClick={handleAddTeam}>
+                                    Add
+                                </Button>
+                            </DialogFooter>
+                        </DialogContent>
+                    </Dialog>
```

Comment:
This dialog (and probably all others) should be moved into a dedicated component to improve readability.